File: 551-student-attendance-record-i/551-student-attendance-record-i.cpp
class Solution
{
public:
    bool checkRecord(string s)
    {
        int ca=0,cl=0,n=s.size();
        bool a=true;
        
        for(int i=0;i<n;i++)
        {
            if(s[i] == 'L')
            {
                cl++;
                if(cl>2)
                    a = false;
            }

            else
            {
                cl=0;
                if(s[i] == 'A')
                {    
                    ca++;
                    if(ca>=2)
                        a = false;
                }
            }
        }
              
        return (a == true);
    }
};

File: 551-student-attendance-record-i/551-student-attendance-record-i.java
// class Solution {
//     public boolean checkRecord(String s) 
//     {
//     		if (s.contains("LLL")) 
//                 return false;
    		
//     		String str = s.replaceAll("A", "");
    		
//     		if ((s.length() - str.length()) > 1)
//     		{
//     			return false;
//     		}
    		
    		
//         return true;
        
//     }
// }
class Solution 
{
    public boolean checkRecord(String s) 
    {
        int countA=0;
        int continuosL = 0;
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i) == 'A')
            {
                countA++;
                continuosL = 0;
            }
            else if(s.charAt(i) == 'L')
                continuosL++;
            else
                continuosL = 0;
            
            if(countA >1 || continuosL > 2 )
                return false;
        }
        return true;

    }
}